SharpCompress/Common/EntryStream.cs

```diff
@@ -26,12 +26,27 @@ public void SkipEntry()
             completed = true;
         }
 
+        public bool Cancelled { get; private set; }
+
+        /// <summary>
+        /// Indicates that the remainder of the stream is not required.
+        /// On dispose, the entry will not be skipped, so it helps with efficiency.
+        /// The downside is that subsequent entries are not usable, as the compressed stream is not positioned at an entry boundary.
+        /// </summary>
+        public void Cancel()

```

Comment:
I don't like this because it does invalidate the Reader altogether.  If you really want to abort reading entries, you should abort the Reader itself.  Not the stream for a single entry in the Reader.